ospd/ospd.py

```diff
@@ -584,35 +569,42 @@ def handle_timeout(self, scan_id: str, host: str) -> None:
             value="{0} exec timeout.".format(self.get_scanner_name()),
         )
 
-    def remove_scan_hosts_from_target_progress(
-        self, scan_id: str, exc_hosts_list: List
-    ) -> None:
-        """ Remove a list of hosts from the main scan progress table."""
-        self.scan_collection.remove_hosts_from_target_progress(
-            scan_id, exc_hosts_list
-        )
-
-    def set_scan_host_finished(
+    def sort_host_finished(
         self, scan_id: str, finished_hosts: Union[List[str], str],
     ) -> None:
-        """ Add the host in a list of finished hosts """
+        """ Check if the finished host in the list was alive or dead
+        and update the corresponding alive_count or dead_count. """
         if isinstance(finished_hosts, str):
             finished_hosts = [finished_hosts]
 
-        self.scan_collection.set_host_finished(scan_id, finished_hosts)
+        alive_hosts = []
+        dead_hosts = []
 
-    def set_scan_progress(self, scan_id: str, progress: int) -> None:
-        """ Sets scan_id scan's progress which is a number
-        between 0 and 100. """
-        self.scan_collection.set_progress(scan_id, progress)
+        current_hosts = self.scan_collection.get_current_target_progress(
+            scan_id
+        )
+        for finished_host in finished_hosts:
+            progress = current_hosts.get(finished_host)
+            if progress == 100:
+                alive_hosts.append(finished_host)
+            if progress == -1:
+                dead_hosts.append(finished_host)
```

Comment:
Would be good to use some constants for the progress here
```python
PROGRESS_FINISHED = 100
PROGRESS_DEAD_HOST = -1
```
because using constants improves the readability a lot e.g.
`if progress == PROGRESS_DEAD_HOST` is easy to reason about.